lib/bulbs-elements/store/store.js

```diff
@@ -42,13 +65,15 @@ function dispatchFunction (store, action, payload) {
     console.time(`dispatch ${action.type}`);
   }
 
-  let nextState = cloneObject(store.state);
-  let actionState = nextState[action.fieldName];
-  let actionNextState = cloneObject(actionState);
-  actionNextState = action(actionNextState, payload, store);
-  nextState[action.fieldName] = actionNextState;
+  let priorState = store.state;
+  let nextState = Object.assign({}, priorState);
+  let priorFieldValue = priorState[action.fieldName];
+  let nextFieldValue = cloneObject(priorFieldValue);
```

Comment:
It is possible for a naughty action to reach into the store to mutate another field's state.
This in particular should not be done:
```js
// An action of the foo field
FooField = {
  initialState: {},
  actions: {
    naughtyAction(state, argument,  store) {
      store.state.bar.baz = 'mutated';
    },
  },
};
```